common/src/main/java/org/apache/falcon/entity/parser/FeedEntityParser.java

```diff
@@ -157,14 +157,30 @@ private void validateLifecycle(Feed feed) throws FalconException {
         LifecyclePolicyMap map = LifecyclePolicyMap.get();
         for (Cluster cluster : feed.getClusters().getClusters()) {
             if (FeedHelper.isLifecycleEnabled(feed, cluster.getName())) {
-                if (FeedHelper.getRetentionStage(feed, cluster.getName()) == null) {
-                    throw new ValidationException("Retention is a mandatory stage, didn't find it for cluster: "
+                if ((FeedHelper.getRetentionStage(feed, cluster.getName()) != null) || (FeedHelper.getArchivalStage(feed, cluster.getName()) != null)) {
+                    validateRetentionStage(feed, cluster);
+                    validateArchivalStage(feed, cluster);
+                    for (String policyName : FeedHelper.getPolicies(feed, cluster.getName())) {
+                        map.get(policyName).validate(feed, cluster.getName());
+                    }
+                } else {
+                    throw new ValidationException("Atleast one of Retention/Archival is a mandatory stage, didn't find it for cluster: "
                             + cluster.getName());
                 }
-                validateRetentionFrequency(feed, cluster.getName());
-                for (String policyName : FeedHelper.getPolicies(feed, cluster.getName())) {
-                    map.get(policyName).validate(feed, cluster.getName());
-                }
+            }
+        }
+    }
+
+    private void validateRetentionStage(Feed feed, Cluster cluster) throws FalconException {
+        if (FeedHelper.getRetentionStage(feed, cluster.getName()) != null) {
+            validateRetentionFrequency(feed, cluster.getName());
+        }
+    }
+
+    private void validateArchivalStage(Feed feed, Cluster cluster) throws FalconException {
+        if (FeedHelper.getArchivalStage(feed, cluster.getName()) != null) {
+            if (FeedHelper.getArchivalStage(feed, cluster.getName()).getLocation().getPath().isEmpty()) {
+                throw new ValidationException("Location path cannot be empty.");
```

Comment:
A little more details on the message like the mention of archival stage for the given cluster will be useful.